engine/src/main/java/io/camunda/zeebe/engine/api/ProcessingScheduleService.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.engine.api;
+
+import io.camunda.zeebe.scheduler.future.ActorFuture;
+import java.time.Duration;
+import java.util.function.BiConsumer;
+
+public interface ProcessingScheduleService {
+
+  void runDelayed(Duration delay, Runnable followUpTask);
+
+  <T> void runOnCompletion(ActorFuture<T> precedingTask, BiConsumer<T, Throwable> followUpTask);
+
+  default void runAtFixedRate(final Duration delay, final Runnable task) {
```

Comment:
:upside_down_face: I would rather inline it to reduce the API of this interface as much as possible, and leave it up to the caller to implement this pattern.

Reply:
I would prefer to have a default implementation in order to have consistent exception handling.